.github/workflows/lint-bpf-checks.yaml

```diff
@@ -85,57 +85,8 @@ jobs:
         env:
           COCCINELLE_HOME: /usr/local/lib/coccinelle
 
-  set_clang_dir:
-    name: Set clang directory
-    runs-on: ubuntu-latest
-    outputs:
-      clang_dir: ${{ steps.set_dir.outputs.clang_dir }}
-    steps:
-    - name: Set directory
-      id: set_dir
-      run: echo "clang_dir=$HOME/.clang" >> $GITHUB_OUTPUT
-
-  # Runs only if code under bpf/ is changed.
-  build_all:
-    needs: [check_changes, set_clang_dir]
-    if: ${{ needs.check_changes.outputs.bpf-tree == 'true' || needs.check_changes.outputs.workflow-description == 'true' }}
-    name: Build Datapath
-    runs-on: ubuntu-22.04
-    steps:
-      - name: Install Go
-        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
-        with:
-          # renovate: datasource=golang-version depName=go
-          go-version: 1.22.2
-      - name: Cache LLVM and Clang
-        id: cache-llvm
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
-        with:
-          path: ${{ needs.set_clang_dir.outputs.clang_dir }}
-          key: llvm-10.0
-      - name: Install LLVM and Clang prerequisites
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libtinfo5
-      - name: Install LLVM and Clang
-        uses: KyleMayes/install-llvm-action@dec985c8d7b46a2f363ea1a78f660c946a3349ea # v2.0.1
-        with:
-          version: "10.0"
-          directory: ${{ needs.set_clang_dir.outputs.clang_dir }}
-          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
-      - name: Checkout code
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        with:
-          persist-credentials: false
-          fetch-depth: 0
-      - name: Build all BPF datapath permutations
-        env:
-          V: 0
-        run: |
-          make --quiet -C bpf build_all || (echo "Run 'make -C bpf build_all' locally to investigate build breakages"; exit 1)
```

Comment:
Wait, why are we removing the step to build the datapath?

Reply:
I guess that this is the short (<10m) result to make sure the BPF code compiles, which doesn't require developers to wait for the full `lint-build-commits.yaml` job to complete to get initial feedback. I think we actually want to retain this step, but switch it to also leverage `contrib/scripts/builder.sh`.